### PR TITLE
Fetching Robots.txt File on 301 Redirect

### LINKED ISSUE #9: Fetch file on 301 for robots.txt base scanning

### PR DESCRIPTION
**Pull request by macdoos**
resolves #9 

Screenshot 2024-06-14 at 16 10 09

### CODE REVIEW

**Comment by vmfunc** [comment]
Hello, thanks for your pull request! However, anything related to robots.txt are to be done in [pkg/scan/scan.go](https://github.com/lunchcat/sif/blob/main/pkg/scan/scan.go)

File: sif.go [hunk 1]
```diff
@@ -60,6 +62,33 @@ func New(settings *config.Settings) (*App, error) {
...
+	}
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return "", err
+	}
+
+	return string(body), nil
+}
+
```
Comment (vmfunc) [inline sif.go:91]: see https://github.com/lunchcat/sif/blob/main/pkg/scan/scan.go

File: sif.go [hunk 2]
```diff
@@ -87,6 +116,12 @@ func (app *App) Run() error {
 		}
 
 		if !app.settings.NoScan {
+			robotsTxt, err := fetchRobotsTXT(fmt.Sprintf("%s/robots.txt", url))
+			if err != nil {
+				log.Errorf("Failed to fetch robots.txt for %s: %v", url, err)
+			} else {
+				log.Infof("robots.txt content for %s:\n%s", url, robotsTxt)
+			}
```
Comment (vmfunc) [inline sif.go:124]: see https://github.com/lunchcat/sif/blob/main/pkg/scan/scan.go

**Review by vmfunc — CHANGES_REQUESTED** [review]

**Comment by macdoos** [comment]
@vmfunc reverted changes to main and modified `scan.go`

File: pkg/scan/scan.go [hunk 3]
```diff
@@ -15,6 +15,24 @@ import (
...
+	resp, err := client.Get(url)
+	if err != nil {
+		return nil, err
+	}
+
+	if resp.StatusCode == http.StatusMovedPermanently {
+		redirectURL := resp.Header.Get("Location")
+		if redirectURL == "" {
+			return nil, fmt.Errorf("redirect location is empty")
+		}
```
Comment (vmfunc) [inline pkg/scan/scan.go:28]: please look into the way we handle logging. do not return an error and use our log package

**Review by vmfunc — CHANGES_REQUESTED** [review]

**Pull request merged**